erpnext/public/js/utils.js

```diff
@@ -629,41 +629,57 @@ erpnext.utils.map_current_doc = function(opts) {
 			frappe.get_meta(items_doctype).fields.forEach(function(d) {
 				if(d.options===opts.source_doctype) link_fieldname = d.fieldname; });
 
-			// search in existing items if the source_name is already set and full qty fetched
-			var already_set = false;
-			var item_qty_map = {};
-
-			$.each(cur_frm.doc.items, function(i, d) {
-				opts.source_name.forEach(function(src) {
-					if(d[link_fieldname]==src) {
-						already_set = true;
-						if (item_qty_map[d.item_code])
-							item_qty_map[d.item_code] += flt(d.qty);
-						else
-							item_qty_map[d.item_code] = flt(d.qty);
-					}
+			if (opts.remove_common_rows) {
+				$.each(cur_frm.doc.items, (parent_idx, parent_val) => {
+					$.each(opts.args.filtered_children, (child_idx, child_val) => {
+						if (parent_val[link_fieldname + "_item"] == child_val) {
```

Comment:
@ankush Here i am using "material_request_item" or "{link_fieldname}_item" to get the row hash.

Reply:
We don't maintain this naming everywhere, hence we need to pass the actual field name 😅 

Check other doctypes like sales invoice for example. The function you're modifying is used by 10+ other doctypes.

Reply:
I understood, Btw It will work fine for the Sale Invoice 😅